test/powershell/Modules/Microsoft.PowerShell.Management/Test-Connection.Tests.ps1

```diff
@@ -3,19 +3,47 @@
 
 Import-Module HelpersCommon
 
+function GetGatewayAddress
+{
+    return [System.Net.NetworkInformation.NetworkInterface]::GetAllNetworkInterfaces() |
+        Where-Object { $_.OperationalStatus -eq 'Up' -and $_.NetworkInterfaceType -ne 'Loopback' } |
+        ForEach-Object { $_.GetIPProperties().GatewayAddresses } |
+        Select-Object -First 1 |
+        ForEach-Object { $_.Address.IPAddressToString }
+}
+
+function GetExternalHostAddress([string]$HostName)
+{
+    if (-not $HostName)
+    {
+        return
+    }
+
+    try
+    {
+        return [System.Net.Dns]::GetHostEntry($HostName).AddressList |
+            Where-Object { $_.AddressFamily -eq 'InterNetwork' } |
+            Select-Object -First 1 |
+            ForEach-Object { $_.IPAddressToString }
+    }
+    catch
+    {
+        return
+    }
+}
+
 Describe "Test-Connection" -tags "CI" {
     BeforeAll {
         $hostName = [System.Net.Dns]::GetHostName()
+        $gatewayAddress = GetGatewayAddress
+        $publicHostAddress = GetExternalHostAddress -HostName $hostName
+
+        $testAddress = if ($publicHostAddress) { $publicHostAddress } else  { $gatewayAddress }
```

Comment:
Not blocking, but could have used the null conditional assignment operator here

Reply:
This needs to port back to 6.2 -- deliberately removed new operators in most recent commit